tests/utils.py

```diff
@@ -839,6 +863,13 @@ def snapshot_context(token, ignores=None, tracer=None, async_mode=True):
         if async_mode:
             # Patch the tracer writer to include the test token header for all requests.
             tracer.writer._headers["X-Datadog-Test-Token"] = token
+
+            # Also add a header to the environment for subprocesses test cases that might use snapshotting.
+            existing_headers = parse_tags_str(os.environ.get("_DD_TRACE_WRITER_ADDITIONAL_HEADERS", ""))
```

Comment:
do we think this is needed? this should only ever be used for tests

Reply:
I think it would be awfully confusing for some poor in the future to see the `_DD_TRACE_WRITER_ADDITIONAL_HEADERS` variable (for whatever reason), set it and be awfully confused as to why their environment variable is being overwritten.

That was the thinking here anyway, did you have a concern @brettlangdon?

Reply:
my only concern is that this env variable should only ever be used by these specific tests no?

wondering if it is worth dealing with this scenario that should never exist/we don't want to support outside of the tests?

not a strong opinion

Reply:
Got it. Yeah I think this is a low effort way to save a potential headache in the future with not much of a drawback so I'm ok with keeping it